### PR TITLE
Hot wallet support

### LINKED ISSUE #49: Hot/Laptop signer (signing without a hardware signing device)
For the first release of Liana, we plan to support the only two hardware signing devices supporting Miniscript: the Ledger and the Specter hardware wallets.

Some people using hot keys on their laptop at the moment might be willing to switch to Liana for, say, the inheritance planning but not be willing to start using a hardware signing device. For this usecase we could implement private key storage in the data directory and signing on the GUI directly without having to connect a signing device.
First, is there such an interest?

Second, if we decide to do it we need to discuss how to approach it. Do we want to encrypt it? Probably? How long do we keep it in memory? How do we mlock the secrets in a portable manner? Is [Rust going to be a footgun](https://benma.github.io/2020/10/16/rust-zeroize-move.html) when managing secrets?

Thoughs?

### PR DESCRIPTION
**Pull request by darosior**
This introduces a new `signer` module. Its purpose is to provide clients of the Liana daemon (such as the GUI) with tools to sign transactions.

For now, the only signer available is a hot signer based on BIP39. It allows to generate new mnemonics from OS-provided randomness (with added randomness from the CPU if available and mixed-in contextual data). Mnemonics are stored in plaintext in a `mnemonics` folder at the root of a provided data directory.

Fixes #49.

### CODE REVIEW

**Comment by darosior** [comment]
I've rebased this on top of #282, implemented a method for signing a PSBT and added unit tests for signing a PSBT with varying number of inputs, using different signers and possibly multiple keys on a single signer. I think this is now ready for review.

**Comment by darosior** [comment]
Rebased on latest #282.

**Comment by darosior** [comment]
Rebased on master.

**Comment by darosior** [comment]
[`ENT`](https://www.fourmilab.ch/random/) gives a pretty decent result for a couple dozen millions of runs of `getrandom`:
```
Entropy = 8.000000 bits per byte.

Optimum compression would reduce the size
of this 912444480 byte file by 0 percent.

Chi square distribution for 912444480 samples is 254.09, and randomly
would exceed this value 50.43 percent of the times.

Arithmetic mean value of data bytes is 127.4952 (127.5 = random).
Monte Carlo value for Pi is 3.141641048 (error 0.00 percent).
Serial correlation coefficient is 0.000022 (totally uncorrelated = 0.0).
```

File: src/lib.rs [hunk 1]
```diff
@@ -7,6 +7,8 @@ mod database;
 pub mod descriptors;
 #[cfg(feature = "jsonrpc_server")]
 mod jsonrpc;
+mod random;
+pub mod signer;
```
[the remark this answers is not in the document]
Comment (darosior) [reply src/lib.rs:11]: Done

**Review by edouardparis — APPROVED** [review]
ACK 20f394a452e425f4bdad0195637612499575a878
tested with GUI

**Pull request merged**